### DISCUSSION TITLE
Kb/Mouse SAM Module on Installation

### DISCUSSION THREAD

**Discussion by hellohhhhgh** [linux-surface/linux-surface#1395] [Q&A]
On Live Media with DE shipped, e.g. FedoraWorkstation or Ubuntu, there we can use Kbd and Touchpad without any installation or modifing. And also when I booted fresh Linux, there arent any problem using Kbd and touchpad.
I thought Surface's Kbd and Touchpad used SAM, which isnt included in upstream kernel, how is it possible??

Also when I try to use kbd in non-live OS, like Fedora Silverblue or Ubuntu nonLive ISO or Archlinux, I cant use my Kbd.
I tried to replace kernel inside installer runtime into kernel-surface, and I checked inside installer, but kbd still doesnt works.

My question is, where does surface keyboards are dependant to? Why it sometimes works sometimes doesnt works?

**Comment by qzed** [comment]
Support should be upstream for almost all of the devices by now. At least on any fairly recent kernel version. Notable exceptions are the touchpad on the Surface Laptop 2 (we don't even have proper support for that here yet) and the ARM-based devices.

So support on ISOs depends on two major things:
- The device in question and the kernel version (i.e. if the kernel is new enough to support the device). Older devices generally already work well with older kernels. Newer devices may require newer kernels.
- Whether the distro developers enabled that support in the config.

**Reply by hellohhhhgh** [reply]
I was just confused by I can use kbd in Fedora Live Image, but I cant use it in Fedora Installer Image (even if I used kernel-surface).. so I thought in Installer Image's kernel it doesnt picks up modules maybe?

**Reply by qzed** [reply]
Possibly. If you can find a way to get and post the kernel config of that image's kernel, I can check if the required options are set.
